tests/test_triangulate.py

```diff
@@ -197,3 +197,238 @@ def test_change_limit():
         limit=0.5,
     )
     assert len(triangles) == 12
+
+
+TEST_POLYGONS = [
+    (np.array(x, dtype=np.float32), y)
+    for x, y in [
+        ([(0, 0), (1, 1), (0, 2), (2, 1)], [(3, 2, 1), (0, 3, 1)]),
+        (
+            [(0, 0), (0, 1), (1, 2), (2, 1), (2, 0), (1, 0.5)],
+            [(4, 3, 5), (3, 2, 1), (5, 3, 1), (5, 1, 0)],
+        ),
+        (
+            [(0, 1), (0, 2), (1, 1.5), (2, 2), (2, 1), (1, 0.5)],
+            [(4, 3, 2), (2, 1, 0), (4, 2, 0), (5, 4, 0)],
+        ),
+        (
+            [(0, 1), (0, 2), (1, 0.5), (2, 2), (2, 1), (1, -0.5)],
+            [(2, 1, 0), (2, 0, 5), (4, 3, 2), (5, 4, 2)],
+        ),
+        ([(0, 0), (1, 2), (2, 0), (1, 1)], [(2, 1, 3), (3, 1, 0)]),
+        ([(0, 0), (0, 1), (0.5, 0.5), (1, 0), (1, 1)], [(3, 4, 2), (2, 1, 0)]),
+        ([(0, 0), (1, 0), (0.5, 0.5), (0, 1), (1, 1)], [(2, 4, 3), (1, 2, 0)]),
+    ]
+]
+
+
+def _renumerate_triangles(polygon, points, triangles):
+    point_num = {tuple(point): i for i, point in enumerate(polygon)}
+    return [
+        tuple(point_num[tuple(points[point])] for point in triangle)
+        for triangle in triangles
+    ]
+
+
+@pytest.mark.parametrize(('polygon', 'expected'), TEST_POLYGONS)
+def test_triangulate_polygons_with_edge_non_convex(polygon, expected):
+    (triangles, points), _edges = triangulate_polygons_with_edge([polygon])
+    triangles_ = _renumerate_triangles(polygon, points, triangles)
+    assert triangles_ == expected
+
+
+def test_triangulate_polygon_in_polygon_numpy():
+    polygons = [
+        np.array([(0, 0), (10, 0), (10, 10), (0, 10)], dtype=np.float32),
+        np.array([(4, 4), (6, 4), (6, 6), (4, 6)], dtype=np.float32),
+    ]
+    (triangles, points), _edges = triangulate_polygons_with_edge(polygons)
+    assert len(points) == 8
+    assert len(triangles) == 8
+
+
+def test_triangulate_polygon_with_hole():
+    polygon = np.array(
+        [
+            (0, 0),
+            (10, 0),
+            (10, 10),
+            (0, 10),
+            (0, 0),
+            (4, 4),
+            (6, 4),
+            (6, 6),
+            (4, 6),
+            (4, 4),
+        ],
+        dtype=np.float32,
+    )
+    (triangles, points), _edges = triangulate_polygons_with_edge([polygon])
+    assert len(points) == 8
+    assert len(triangles) == 8
+
+
+def test_triangulate_polygon_segfault1():
```

Comment:
```suggestion
def test_triangulate_polygon_causes_segfault1():
```

Reply:
Not sure why these tests are numbered 1, 2, 3, 4, 5 Could we parameterize these test cases by setting the polygons to constants and then run them through 1 test?

Reply:
These are five polygons that cause segfault when testing in C++ implementation. These are in separate functions as it simplify debugging. 
Also, in rust implementation they were useful to finding bugs in implementation.

Reply:
Sounds good!